avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastDeserializerGenerator.java

```diff
@@ -606,68 +609,61 @@ private void processArray(JVar arraySchemaVar, final String name, final Schema a
     }
     JInvocation finalNewArrayExp = newArrayExp;
 
-    JExpression chunkLengthGreaterThanZero = chunkLen.gt(JExpr.lit(0));
 
-    ifCodeGen(parentBody, chunkLengthGreaterThanZero, then1 -> {
```

Comment:
FYI: this looks like a big change but it's mostly just a move and an indentation change. The list instantiation (and potential reuse) now happens first, and the `if chunk length > 0` check as well as its else block is now removed. Then most of the remaining code is just an indentation change...